ladybug/_datacollectionbase.py

```diff
@@ -67,24 +73,17 @@ def header(self):
     @property
     def datetimes(self):
         """Return datetimes for this collection as a tuple."""
-        return tuple(self._datetimes)
+        return self._datetimes
 
     @property
     def values(self):
-        """Return the Data Collection's list of numerical values."""
+        """The Data Collection's list of numerical values."""
         return tuple(self._values)
```

Comment:
Good question.  So, even when the collection is mutable, we still need the result returned by `values` to be immutable because, if it returned a list, people would be able to append items to the list and then suddenly the number of values no longer matches the number of datetimes on the collection.  So I think we agree that the otuput here needs to be a tuple.

However, if we want to support people being able to set individual values of the collection by index by using `__setitem__`, then this gets very complicated when the underlying values are in a tuple and not a list.  Basically, every time that `__setitem__` gets called, we have to turn the underling values into a list, then change the single item to be the value that came through `__setitem__` and then we have to convert all of the values back to a tuple.  Such complication essentially destroys workflows where people initialize a Data Collection with dummy values and then set each item to be what they want.  If we want to get rid of the `__setitem__` capability, we can change the underlying values to be a tuple.  That way the only way people can edit a mutable collection is by using the values setter after building up a list of values.  This would make me a bit sad, though, since I really like the `__setitem__` workflow and I think that you do too.

Reply:
> So, even when the collection is mutable, we still need values to return a tuple because, if it returned a list, people would be able to append items to the list

I see your point and making it a tuple make it harder but for what it worth you can add items to tuples without making it a list. try:  `(1, 2, 3) + (4, 5)`